src/Workspaces/Core/Portable/Rename/RenameLocation.ReferenceProcessing.cs

```diff
@@ -276,25 +276,29 @@ private static string TrimNameToAfterLastDot(string name)
                 }
 
                 // If we're renaming a named type, we'll also have to find constructors and
-                // destructors declarations that match the name
+                // destructors declarations that match the name if we're in C#. In VB these
+                // methods are not named to match the class.
                 if (referencedSymbol.Kind == SymbolKind.NamedType)
                 {
                     var namedType = (INamedTypeSymbol)referencedSymbol;
-
-                    foreach (var method in namedType.GetMembers().OfType<IMethodSymbol>())
+                    if (namedType.Language == LanguageNames.CSharp)
```

Comment:
This is the only actual change. This may be contentious. Other options include adding `StructorNamesMatchTypeName` (working name) to ISyntaxFactsService and passing that in. I can't check the Kind because it's language-neutral. Other suggestions welcome...